azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/devicecode/DeviceCodeAccount.java

```diff
@@ -47,6 +50,8 @@ protected TokenCredential createTokenCredential() {
         if (executorService.isShutdown()) {
             throw new AzureToolkitAuthenticationException("Cannot device login twice.");
         }
+        AzureEnvironment env = Azure.az(AzureCloud.class).getOrDefault();
+        this.entity.setEnvironment(env);
```

Comment:
Do we need to call `AzureEnvironmentUtils.setupAzureEnvironment(env);` like OAuth?

Reply:
AzureEnvironmentUtils.setupAzureEnvironment will be handled in another feature: use az cloud to avoid duplicate parameter azure env.